zproject/backends.py

```diff
@@ -120,6 +133,31 @@ def get_full_name(self, *args, **kwargs):
         # type: (*Any, **Any) -> Text
         raise NotImplementedError
 
+    def get_authenticated_user(self, *args: Any, **kwargs: Any) -> Optional[UserProfile]:
+        raise NotImplementedError
+
+    @handle_http_errors
+    def do_auth(self, *args: Any, **kwargs: Any) -> Optional[HttpResponse]:
+        """
+        This function is called once the authentication workflow is complete.
+        We override this function to:
+            1. Inject `return_data` kwarg. This will be used by
+               `authenticate()` function to make the decision.
+            2. Call the proper authentication function to get the user.
+
+        The actual decision on authentication is done in
+        SocialAuthMixin._common_authenticate().
+
+        SocialAuthMixin.get_authenticated_user is expected to be overridden by
+        the derived class to add custom logic for authenticating the user and
+        returning the user.
+        """
+        kwargs['return_data'] = {}
+        # `strategy` comes from Python Social Auth.
+        kwargs['realm_subdomain'] = get_subdomain(self.strategy.request)  # type: ignore
+        user_profile = self.get_authenticated_user(*args, **kwargs)
+        return self.process_do_auth(user_profile, *args, **kwargs)
```

Comment:
Can you extract this (+ the `GithubAuthBackend` update required by this) to an atomic commit that works at the base of the branch?

Reply:
Yep.